### PR TITLE
Denys sikorskyi

### PR DESCRIPTION
**Pull request by Dsikora369**
<h1>Description</h1>
Made first task with 80% accuracy and second with 0.11 loss.

- [x] Executed code few times from start to end without crashing
- [ ] Only ran once
- [x] Code have dedicated unit tests

### CODE REVIEW

File: Dsikora369_code/task_1/titanic.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    ['CategoricalFare'], as_index=False).mean()
+
+"""Also, it is very useful to notice whether passanger had a cabin or not and which cabin.
+
+"""
+
+
+def get_cabin(cabin):
+    for i in str(cabin):
+        if i in ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'Z']:
```
Comment (meanalexrin) [inline Dsikora369_code/task_1/titanic.py:71]:
```suggestion
        if i in 'ABCDEFGHZ':

```

Python rows is arrays on itself, so you may not split them letter-by-letter

File: Dsikora369_code/task_1/titanic.py [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+def get_title(name):
+    pattern = r'([A-Za-z]+)\.'
+    title_search = re.search(pattern, name)
+    if title_search:
+        return title_search.group(1)
+    return ""
+
+
+def get_cabin(cabin):
```
Comment (meanalexrin) [inline Dsikora369_code/task_1/titanic.py:147]: This function is already declared above

File: Dsikora369_code/task_1/titanic.py [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+drop_elements = ['PassengerId', 'Name', 'Ticket', 'Cabin', 'SibSp',
+                 'Parch', 'FamilySize']
+
+train = train.drop(drop_elements, axis=1)
+train
+
+"""Creating function for data preprocessing and dividing into training and validation data."""
+
+
+def get_title(name):
```
Comment (meanalexrin) [inline Dsikora369_code/task_1/titanic.py:139]: This function is already declared above

File: Dsikora369_code/task_1/titanic.py [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+
+"""Column 'Embarked' has NaNs that will be replced with median."""
+
+train['Embarked'] = train['Embarked'].fillna(train['Embarked'].mode()[0])
+
+"""Adding Ageclass feature.
+
+"""
+
+train['Age*Class'] = train.Age * train.Pclass
```
Comment (meanalexrin) [inline Dsikora369_code/task_1/titanic.py:125]: You may consider doing such a transformation automatically to many features using `PolynomialFeatures`
https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.PolynomialFeatures.html

File: Dsikora369_code/task_1/titanic.py [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+        )
+    )
+)
+
+"""**Creating model**"""
+model = tf.keras.Sequential([
+    Dense(12, activation=tf.nn.leaky_relu, input_shape=[9]), Dropout(0.2), 
+    Dense(5, activation=tf.nn.leaky_relu), Dense(2, activation=tf.nn.softmax)])
+opt = tf.keras.optimizers.Adam(
+    learning_rate=0.01, beta_1=0.9, beta_2=0.999, epsilon=1e-07, amsgrad=True,
```
Comment (meanalexrin) [inline Dsikora369_code/task_1/titanic.py:260]: Note that if you are using default function arguments you can left them unspecified 
https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Adam

File: Dsikora369_code/task_1/titanic.py [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+              loss=tf.keras.losses.SparseCategoricalCrossentropy(),
+              metrics=['accuracy'])
+
+num_train_examples = [i for i, _ in enumerate(training_dataset)][-1] + 1
+num_val_examples = [i for i, _ in enumerate(validation_dataset)][-1] + 1
+BATCH_SIZE = 256
+train_dataset = training_dataset.cache().repeat().shuffle(
+    num_train_examples).batch(BATCH_SIZE)
+val_dataset = validation_dataset.cache().batch(BATCH_SIZE)
+model.fit(train_dataset, validation_data=val_dataset, epochs=1000,
```
Comment (meanalexrin) [inline Dsikora369_code/task_1/titanic.py:273]: `epochs=1000` is should be quite an overkill for this task with 7 hundred samples :D

File: Dsikora369_code/task_2/house_prices.py [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+randomized_search_result = final_model.randomized_search(grid, X=X_train,
+                                                         y=y_train,
+                                                         verbose=False,
+                                                         plot=True)
+
+"""Final model."""
+
+print("\n========================================================")
+print(" Results from Grid Search ")
+print("========================================================")
```
Comment (meanalexrin) [inline Dsikora369_code/task_2/house_prices.py:242]: You may use `'='*40` to create long repetitive rows

**Review by meanalexrin — DISMISSED** [review]
LGTM - Looks Good To Me :+1:

**Comment by Dsikora369** [comment]
Fixed
